controllers/serving/inference_controller.go

```diff
@@ -273,6 +294,80 @@ func (ir *InferenceReconciler) syncTrafficDistribution(inf *servingv1alpha1.Infe
 	return nil
 }
 
+// syncPredictorTrafficDistribution syncs traffic-routing resources to ensure traffic is distributed by the
+// user configuration required.
+// For example, user A serves two models and distribute 90% traffic to model A.1 and the left to
+// model A.2, the network topology looks like this:
+//
+// User
+//  |---request---> VirtualService
+//                        |--- 90% ---> Deploy-Of-Model-A.1
+//                        |--- 10% ---> Deploy-Of-Model-B.1
+func (ir *InferenceReconciler) syncPredictorTrafficDistribution(inf *servingv1alpha1.Inference, index int, predictor *servingv1alpha1.PredictorSpec) error {
+	igInCluster := networkingv1beta1.Ingress{}
+	igExists := true
+	igName := genPredictorName(inf, predictor)
+	err := ir.client.Get(context.Background(), types.NamespacedName{Namespace: inf.Namespace, Name: igName}, &igInCluster)
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
similarly, these two if conditions can be combined into a single if err != nil {xxx}. block